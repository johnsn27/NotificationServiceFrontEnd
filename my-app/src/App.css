:root {
  --image-size: 200px;
}

html {
  height:100%;
}

h1 {
  margin:0;
}

.StickyBar {
  display: flex;
}

.StickyBarContents {
  width: 100%;
}

.StickyBarContentsTitle {
  text-align: center;
}

.pageContent {
  width: 100%;
  height: fit-content;
}

.pageContentBelowSticky {
  margin-top: 62px;
  width: 100%;
  height: 474px;
}

.igm-btn {
  width: 100%;
  height: 100%;
}

.leftSide {
  float: left;
  margin-left: 5px;
  width: 10%;
  height: 100%;
  justify-content: space-evenly;
  flex-direction: column;
  align-items: center;
  min-width: 100px;
}

.rightSide {
  float: right;
  width:85%;
}
.imageContent {
  width: 100%;
  height: 100%;
}

.image {
  width:100%;
  height: 40%;
}

.imageCaptionText {
  text-align: center;
  height: 20%;
}

.buttonCollection {
  height: 20%;
  position: relative;
}

.mySectionButton {
  width: 94%;
  height: 29.3%;
  margin: 3%;
}

.roomsButton {
  width: 94%;
  height: 29.3%;
  margin: 2%;
}

.watchButton {
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  height: 45%;
}

.bookButton {
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  position: absolute;
  width: 100%;
  height: 45%;
  bottom: 0px;
}

.textarea {
  width: 80%;
}

.roomDetailsContents {
  display: flex;
  border-style: solid;
  width: 98%;
  height: 100%;
  margin-right: 5px;
  float: left;
  /* min-width: 1000px; */
}

.roomDetailsContentsLeft {
  width: 20%;
}

.roomDetailsContentsRight {
  width: 75%;
}

.roomDetailsContentsRightInner {
  float: right;
  width: 90%;
}

.textbox {
  display: flex;
  text-align: center;
  width: 100%;
}

.label {
  width:20%;
  text-align: left;
}

.Rooms {
  float: left;
  width: 100%;
  height: 45%;
  position: relative;
}

.RoomsTitle {
  float: left;
}

.RoomsButtons {
  float: left;
  width: 100%;
  height: 100%;
}

.MySection {
  float: left;
  width: 100%;
  height: 45%;
  margin-bottom: 55px;
}

.MySectionTitle {
  float: left;
}

.MySectionButtons {
  float: left;
  width: 100%;
  height: 100%;
  display: flex;
  display: inline;
}

.buttonBorder {
  height: 85%;
  border-style: solid;
  min-width: 106px;
}

.App-link {
  color: #61dafb;
}

.gel-icon {
  height:15px;
  width:20px;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
