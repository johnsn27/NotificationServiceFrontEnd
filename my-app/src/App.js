import React, { Component } from 'react';
import './App.css';
import Btn from '@bbc/igm-btn';
import StickyBar from '@bbc/igm-sticky-bar'
import '@bbc/igm-sticky-bar/dist/StickyBar.css';


function onClick() {
  var element = document.getElementsByClassName("css-xp4uvy select__single-value")[0].innerHTML;
  console.log(element);
  return element;
}

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header"></header>
        <div className="content">
          <StickyBar id="ViewRoom" className="StickyBar">
            <div className="StickyBarContents">
              <div className="StickyBarContentsTitle">
                View Room
              </div>
            </div>
          </StickyBar>
          <div className="leftSide">
            <div className="MySection">
              <div className="MySectionTitle">
                My Section:
            </div>
              <div className="MySectionButtons">
                <div>
                  <Btn
                    type="link"
                    tab-index="1"
                    className="Button"
                    onClick={onClick}
                  >
                    <span>My Account</span>
                  </Btn>
                </div>
                <div>
                  <Btn
                    type="link"
                    tab-index="1"
                    className="Button"
                    onClick={onClick}
                  >
                    <span>My Bookings</span>
                  </Btn>
                </div>
                <div>
                  <Btn
                    type="link"
                    tab-index="1"
                    className="Button"
                    onClick={onClick}
                  >
                    <span>My Watched Rooms</span>
                  </Btn>
                </div>
              </div>
            </div>

            <div className="Room">
              <div className="RoomsTitle">
                Rooms:
            </div>
              <div className="RoomsButtons">
                <div>
                  <Btn
                    type="link"
                    tab-index="1"
                    className="Button"
                    onClick={onClick}
                  >
                    <span>Book a Room</span>
                  </Btn>
                </div>
                <div>
                  <Btn
                    type="link"
                    tab-index="1"
                    className="Button"
                    onClick={onClick}
                  >
                    <span>Search a Room</span>
                  </Btn>
                </div>
                <div>
                  <Btn
                    type="link"
                    tab-index="1"
                    className="Button"
                    onClick={onClick}
                  >
                    <span>Meeting Rooms</span>
                  </Btn>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default App;
