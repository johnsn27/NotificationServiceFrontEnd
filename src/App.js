import React, { Component } from 'react';
import './App.css';
import Btn from '@bbc/igm-btn';
import Dropdown from '@bbc/igm-dropdown-select'

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
        <div className="    ">
          <Btn
            type="link"
            tab-index="1"
            className="Button"
          >
          <span>Watch Room</span>
        </Btn>
        </div>
        </header>
      </div>
    );
  }
}

export default App;
