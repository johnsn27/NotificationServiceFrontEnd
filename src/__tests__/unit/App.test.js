import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import App from '../../App';

describe('App', () => {
  configure({ adapter: new Adapter() });

  it('renders without crashing', () => {
    shallow(<App />);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('contains a button', () => {
  const app = shallow(<App/>);
  expect(app.find('.Button').exists()).toBe(true);
  });
});
