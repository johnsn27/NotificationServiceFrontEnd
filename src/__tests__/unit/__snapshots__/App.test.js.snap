// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="    "
    >
      <button
        className="igm-btn igm-btn--link Button"
        disabled={false}
        tab-index="1"
        type="button"
      >
        <span>
          Watch Room
        </span>
      </button>
    </div>
  </header>
</div>
`;
